### PR TITLE
Fix inputs losing their autofocus when navigating from SlackSheet to non-SlackSheet

### PR DESCRIPTION
**Pull request by mikedemarais**
oh, it works

Fixes RAI-632 https://linear.app/rainbow/issue/RAI-632/cool-modals-when-navigating-from-slack-sheet-to-non-native-stack-input

### CODE REVIEW

File: src/components/expanded-state/WalletProfileCreator.js [hunk 1]
```diff
@@ -106,11 +109,11 @@ export default function WalletProfileCreator({
...
   }, [color]);
 
-  const handleFocusInput = useCallback(async () => {
-    if (inputRef) {
-      inputRef.current.focus();
-    }
-  }, []);
+  const [handleDidFocus] = useMagicAutofocus(inputRef.current);
+  const handleTriggerFocusInput = useCallback(
+    () => inputRef?.current?.focus(),
```
Comment (jinchung) [inline src/components/expanded-state/WalletProfileCreator.js:114]: ooh optional chaining :D

**Comment by jinchung** [comment]
Videos are on slack but wanted to mention:
1. The modal for notifying an action is only available if they import a seed phrase, disappears too quickly
2. Opening up the multiwallet list autoscrolls to the right place then does an extra jump
3. Scrolling around on the multiwallet list gets funny

File: src/components/expanded-state/WalletProfileCreator.js (hunk 1 above)
Comment (mikedemarais) [reply src/components/expanded-state/WalletProfileCreator.js:114]: @wojtus7 taught me

**Comment by mikedemarais** [comment]
> Videos are on slack but wanted to mention:
> 
> 1. The modal for notifying an action is only available if they import a seed phrase, disappears too quickly
> 2. Opening up the multiwallet list autoscrolls to the right place then does an extra jump
> 3. Scrolling around on the multiwallet list gets funny

@jinchung hey!

i just fixed `1.` with this commit https://github.com/rainbow-me/rainbow/pull/708/commits/1741231e2af5d8c9374da3895025bcfe939ac62e

but re: `2.` and `3.` -- i cant replicate this, and also i super duper dont think this PR creates any regressions to that effect, like rly the only way this PR effects `multiwallet`'s render cycle is the addition of the `goBack` dependency here https://github.com/rainbow-me/rainbow/pull/708/files#diff-08c91df5bac56f92dde329260607662aR304 but that should super definitely not be causing the problems you are seeing. would you mind clearing cache or whatever and confirming this is definitely a reproducible issue for you?

**Comment by christianbaroni** [comment]
@mikedemarais I'm seeing #2 and #3 also, only in this branch

**Comment by mikedemarais** [comment]
ok @christianbaroni @jinchung this commit should fix `2.` and `3.` https://github.com/rainbow-me/rainbow/pull/708/commits/aee80c6c11771c05633b7f8183b1607465fe4fe6

**Comment by jinchung** [comment]
> ok @christianbaroni @jinchung this commit should fix `2.` and `3.` [aee80c6](https://github.com/rainbow-me/rainbow/commit/aee80c6c11771c05633b7f8183b1607465fe4fe6)

Confirmed 1, 2, 3 (and the original fixes for input focusing) work!

**Pull request merged**